### PR TITLE
Project Blog - updates #41

### PR DESCRIPTION
**Pull request by kmcnellis**
updates #41 , addresses #30

### CODE REVIEW

**Comment by seveibar** [comment]
@agundy can you verify this is working?

File: client/app/projects/blog/blog.controller.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+angular.module('observatory3App')
+.controller('ProjectsBlogCtrl', function ($scope, $http, $stateParams) {
+    $scope.load = function() {
```
Comment (agundy) [inline client/app/projects/blog/blog.controller.js]: Is this something that needs to be run like this? Not sure the function is needed.

File: server/routes.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ var errors = require('./components/errors');
 module.exports = function(app) {
 
   // Insert routes below
+  app.use('/api/posts', require('./api/post'));
```
Comment (agundy) [inline server/routes.js]: Slightly nitpicking but not sure this is the best endpoint name. Would much rather have a blog endpoint and have posts under that.

**Comment by agundy** [comment]
Works except you can start writing a post without being authenticated. This should not work until you login.

File: server/api/post/post.controller.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+  req.body.author = req.user._id;
+  if (!req.body.projectId){ return res.status(400).send("Project not set"); }
+  Project.findOne({'_id': req.body.projectId }, function (err, project) {
+    if(err) { return handleError(res, err); }
+    if(!project) { return res.status(404).send("Project not found"); }
+
+    // Only someone who is part of the project can write a blog post
+    User.findById(req.user._id, function(err, user) {
+      console.log("user",user);
```
Comment (agundy) [inline server/api/post/post.controller.js]: Shouldn't be in the commit

File: client/app/projects/blog/blog.controller.js (hunk 1 above)
Comment (kmcnellis) [inline client/app/projects/blog/blog.controller.js]: The function reloads the blog posts after sending a new one in

File: server/routes.js (hunk 2 above)
Comment (kmcnellis) [inline server/routes.js]: I agree - @jfucci might have some insight?

File: server/api/post/post.controller.js (hunk 3 above)
Comment (kmcnellis) [inline server/api/post/post.controller.js]: What do you mean?

File: client/app/projects/blog/blog.controller.js (hunk 1 above)
Comment (agundy) [inline client/app/projects/blog/blog.controller.js]: Okay cool. That makes sense. Just wasn't sure and didn't look at it long enough to realize that.

File: server/api/post/post.controller.js (hunk 3 above)
Comment (agundy) [inline server/api/post/post.controller.js]: Console logginging on the serve isn't necessary.

Comment (kmcnellis) [inline server/api/post/post.controller.js]: Slipped through -  I'll fix it

File: server/api/post/post.controller.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    return res.json(post);
+  });
+};
+
+// Creates a new post in the DB.
+exports.create = function(req, res) {
+
+  req.body.author = req.user._id;
+  if (!req.body.projectId){ return res.status(400).send("Project not set"); }
+  Project.findOne({'_id': req.body.projectId }, function (err, project) {
```
Comment (agundy) [inline server/api/post/post.controller.js]: This works but for future reference findById is built into Mongoose and does the same query with a little less code and arguably more clarity. http://mongoosejs.com/docs/api.html#model_Model.findById

**Comment by agundy** [comment]
Thanks a bunch @kmcnellis and @jfucci! Merged.

**Pull request merged**